### PR TITLE
clock: change "add" of negative minutes to "subtract" of positive minutes

### PR DESCRIPTION
**Pull request by MattLewin**
I have read through and attempted to follow all your test modification, contribution, and PR guidelines. Please accept my apology if I booted it somehow.

When working on this exercise in Ruby, I found that there was no need to write subtraction operation _per se_, because the test suite was adding negative numbers instead of subtracting positive ones.

I updated the JSON data to force the exercise to require subtraction, as stated in the [README](https://github.com/exercism/problem-specifications/blob/master/exercises/clock/description.md). 

Per [Test Data Versioning](https://github.com/exercism/problem-specifications#test-data-versioning), because I renamed a property, this qualifies as a MAJOR version change, so I updated the version to 2.0.0. Honestly, though, this feels like a MINOR revision to me. Obviously, you guys know better how this ought to be handled.

### CODE REVIEW

**Comment by rpottsoh** [comment]
Please review #740 starting [here](https://github.com/exercism/problem-specifications/pull/740/files#r109659884).  You'll see there was quite a discussion regarding adding negative numbers or having a subtract method.  This PR will be picking up where #740 left off.

**Comment by petertseng** [comment]
Ideally, what should the test suite test?

Given the https://github.com/exercism/problem-specifications/blob/master/exercises/clock/description.md tells us we should be able to subtract. If we assume that every capability listed by the specification must correspond to some function on the clocks we create, we would thus want to test subtract.

It would be acceptable to leave it as adding a negative (possibly renaming it to `adjust`) and leave it to the one who ports this exercise to a given track to realize that this should be converted to a subtraction, but it seems better to make the intent clear and to make it easy to do the right thing.

It is true that a student implementing a solution has the option of writing subtract as add composed with negate, but I don't think this detail needs to be made apparent in the canonical data.

The alternative interpretation of the problem description is that the implementor only need provide some set of functions with which a sufficiently clever user would be able to add/subtract minutes; and this requirement is satisfied by only having `add`.

What to do depends on what interpretation to take.

**Comment by ErikSchierboom** [comment]
> What to do depends on what interpretation to take.

My interpretation is that we should indeed add a separate `subtract` method. Whether a user solves it internally by calling `add` is then an implementation detail. Splitting the `add` and `subtract` functions is better design IMHO, as it doesn't mix concepts (even though those concepts are closely related).

**Comment by MattLewin** [comment]
@petertseng and @ErikSchierboom's comments reflect my thinking. Given that exercism is meant to be a learning tool, it seemed to me that it should abide by the [Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) to the greatest degree possible. It would be a stretch to say I was astonished that the test cases with names including "subtract" were actually addition operations, but I *was* very surprised.

**Comment by petertseng** [comment]
> `comments` may also need to be revised along with this change. e.g. `adding a positive or negative number of minutes.`

Indeed. Currently it gives roughly equal standing to the two possible approaches (be able to add add a positive and negative number OR have add and subtract functions), possibly slightly more prominence to the former approach because it comes first. Since we are having the data strongly suggest the latter approach, having the comments be consistent with that will avoid confusion

File: exercises/clock/canonical-data.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "exercise": "clock",
-  "version": "1.0.1",
+  "version": "2.0.0",
   "comments": [
     "Most languages require constructing a clock with initial values,",
     "adding a positive or negative number of minutes, and testing equality",
```
Comment (rpottsoh) [inline exercises/clock/canonical-data.json]: Should the comment be reworded since the proposed changes preclude the possibility of ever adding (or subtracting) negative minutes?

Comment (rpottsoh) [reply exercises/clock/canonical-data.json]: Sorry, my comment appears to be redundant...

**Comment by MattLewin** [comment]
I was attempting to follow the existing pattern and minimize the change set. I do think it’s intuitive as written: after all, I was easily able to go from zero knowledge to PR quickly. 

With that said, I support whatever you guys think. 

If you do settle on something like “minutesToSubtract,” it would make sense to update the addition cases to “minutesToAdd.”

File: exercises/clock/canonical-data.json [hunk 2]
```diff
@@ -221,66 +221,66 @@
       "cases": [
         {
           "description": "subtract minutes",
-          "property": "add",
+          "property": "subtract",
           "hour": 10,
           "minute": 3,
-          "add": -3,
+          "subtract": 3,
```
[the remark this answers is not in the document]
Comment (kotp) [reply exercises/clock/canonical-data.json]: Since it matches the pattern for `add` I think it is OK.

[the remark this answers is not in the document]
Comment (rpottsoh) [reply exercises/clock/canonical-data.json]: The `"property":` is  `"subtract"`, why does there have to be a value called `"subtract":` as well?  Since the README states that we are adding / subtracting minutes I think `"subtract":` and `"add":` should be called `"amount":` or `"value":`.

**Comment by coriolinus** [comment]
Or `quantity`, if we're bikeshedding.

I do like the idea of putting the operation in the property and maintaining
a consistent input type across different tested properties.

**Comment by MattLewin** [comment]
I’m sorry, did someone say, “bikeshedding?” ;)

**Comment by ErikSchierboom** [comment]
> Or `quantity`, if we're bikeshedding.

I really like this suggestion!

**Comment by MattLewin** [comment]
Honestly, I, too, like the idea of the property dictating the operation, and then using a neutral term such as, “quantity.” Though, to continue the bikeshedding, “value” provides more flexibility. 

I’m not in front of my computer, so I’m uncertain whether I remember the property name correctly, but for the “create” property, it could be changed to "compareTo,” and then “value” would apply to add, subtract, and compareTo. 

This style would also provide a good template/precedent for future exercises.

**Comment by rpottsoh** [comment]
The current list of properties is:
create
add
subtract
equal

**Comment by MattLewin** [comment]
Actually, I take rescind my most recent suggestion. Changing `create` to `compareTo` is unnecessary because we still need `expected` in all cases.

**Comment by petertseng** [comment]
In a [hugely controversial](https://www.youtube.com/watch?v=TIEleCEtqzA) move I am merging this since it has been Approved for a long time (and is not submitted by someone who could merge it themselves).

This PR is a single commit that performs a well-understood change. There are potential follow-up changes, but even if they were added to this PR they would be required to be in a separate commit, since it is a change that does a different thing. So by merging this PR now, I'm not increasing the number of commits by any more than necessary!

**Comment by petertseng** [comment]
Oh, I forgot I mentioned the comment ("positive and negative numbers"), now I need to think about whether I would Request Changes for that. I'll say no, because it says "most languages" require something, so for all I know, it is a truthful statement.
